### DISCUSSION TITLE
Bad performance with mutations

### DISCUSSION THREAD

**Discussion by dominikager** [pimcore/pimcore#10000] [General]
Hello,

does anybody have performance issues with mutations in datahub?
Most of our mutations need 2-3 seconds for a response and they don't have a big logic or similar.
It is only updating an data object.

We already implemented the cache, but that's only for queries.

thanks already for a reply :)

**Comment by dlhck** [comment]
@brusch this is a big topic. Datahub initializes the config for every request - there is basically no caching or registry for that. Maybe there is a way to dump the config with resolver information into a php cache file or something like that, to reduce the overhead for every request.

**Reply by LouKordos** [reply]
I am doing a bulk mutation (not sure if this is an actual bulk mutation, it's a request body with many `updateProdukt` functions) and it takes ~4500 seconds for 4000 mutations. This can't be right?
